### PR TITLE
un-skip history test and fix golden mismatches

### PR DESCRIPTION
**Pull request by imjasonh**
Signed-off-by: Jason Hall <jason@chainguard.dev>

As discovered in https://github.com/docker/cli/pull/3779#discussion_r974281026, these tests were skipped because CI doesn't run in UTC. The failures don't seem to require UTC, only updating the golden file's column spacing to match current reality.

**- What I did**

Un-skipped tests in history_test.go

**- How I did it**

Removed `skip` from `history_test.go`, added `t.Run` wrappers so that failed assertions didn't stop other tests from running. Updated golden files to match test expectations.

**- How to verify it**

`make dev && make test-unit`

**- Description for the changelog**

Image history tests are no longer skipped on CI.

**- A picture of a cute animal (not mandatory but encouraged)**

octoeyes

cc @thaJeztah

### CODE REVIEW

File: cli/command/image/history_test.go
Comment (thaJeztah) [inline cli/command/image/history_test.go:94]: <3 subtests

I think it'd be good to capture the `tc` variable in the loop here, in case we'd ever decide to use `t.Parallel()`;

```suggestion
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
```

File: cli/command/image/history_test.go
Comment (thaJeztah) [inline cli/command/image/history_test.go:52]: Tried this locally, and looks like we may have the same issue as in the other test; the output is formatted for the local timezone, which may not be UTC, in which case the `.golden` files won't match;

```bash
=== RUN   TestNewHistoryCommandSuccess/non-human
    history_test.go:102: assertion failed:
        --- expected
        +++ actual
        @@ -1,3 +1,3 @@
        -IMAGE     CREATED AT             CREATED BY   SIZE      COMMENT
        -abcdef    2017-01-01T12:00:03Z   rose         0         new history item!
        +IMAGE     CREATED AT                  CREATED BY   SIZE      COMMENT
        +abcdef    2017-01-01T13:00:03+01:00   rose         0         new history item!

```


Not sure what the best solution is for this one though (don't think the `.golden` files allow for placeholders and/or fuzzy matching 🤔

File: cli/command/image/history_test.go
Comment (imjasonh) [reply cli/command/image/history_test.go:95]: I'm not sure this is strictly needed with `t.Parallel`, but I also don't think it hurts either.

File: cli/command/image/history_test.go
Comment (imjasonh) [reply cli/command/image/history_test.go:52]: I'm running these tests in the dev env on macOS (`make dev`), which seems to set the timezone to UTC. Can we have tests run in UTC so they pass consistently?

(I'm also fine with just continuing to skip this test under non-UTC, but we'd still need to fix it so it _passes_ under UTC)

Comment (thaJeztah) [reply cli/command/image/history_test.go:52]: Ah, yes, correct; running these tests inside the dev environment will run them inside a container (which default to use UTC). Running them directly on the host may not be the "common" way to run them, but (e.g.) to run a "one-off" test from within an IDE may be run that way.

I would suggest to either add back the skip for the whole test, or (now that we have subtests) to add a skip for that specific test (although I'd be fine with just adding back the `t.Skip` if we don't want to complicate things too much).

**Comment by thaJeztah** [comment]
Looks like the second commit is missing a DCO sign-off; when you're updating, could you squash the commits, and fix the sign-off to make the DCO check happy?

Thanks again! Sorry for the delay (had a lot of catching up to do on notifications 😅)

File: cli/command/image/history_test.go
Comment (imjasonh) [reply cli/command/image/history_test.go:52]: So it turned out that even inside `make dev` these were being skipped, even though the dev env's timezone was UTC. 🤔 

I changed the semantics of `notUTCTimezone` to check for an offset != 0, which means these run now in the dev env.

https://go.dev/play/p/9o5SSnHUONR

Regardless, whether we skip the test case or not, I think the bigger issue is that it gets skipped in CI, so there's no safeguard against bugs creeping in for non-UTC users (statistically, most of them 🌎). So I'd like to see if we can make this unskipped in CI somehow at least.

**Comment by imjasonh** [comment]
Friendly ping. Anything else you'd like to see?

**Comment by imjasonh** [comment]
I'm not sure why DCO isn't reporting. I'll try closing/reopening and/or re-pushing the change to see if that wakes it up.

**Comment by imjasonh** [comment]
Another friendly ping. I'm not in any rush to get this done, I just keep noticing it's still open. Anything else blocking?

**Review by vvoland — APPROVED** [review]

**Comment by imjasonh** [comment]
> Another friendly ping. I'm not in any rush to get this done, I just keep noticing it's still open. Anything else blocking?

Another one. I'm also fine just dropping this if there isn't interest in testing this. If that's the case though, I can also just delete the test.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
let me close/reopen to have  a fresh run of CI 🙈

**Comment by thaJeztah** [comment]
Whoops; looks like I never pressed the "merge" button on this one.

I was just working on a similar test, and it looks like there's an workaround that just never occurred to me; adding a `t.SetEnv("TZ", "UTC")` to the test will make the test use UTC. I'll do a quick follow-up after this one's merged 👍

**Pull request merged**
